### PR TITLE
[IPC/Zenoh] add type info to message metadata for publisher

### PR DESCRIPTION
**Pull request by filippobrizzi**
# Description
Zenoh allows to attach metadata to every message, we already use it to share the sequence number and session id. Now it also pass the type info. 

In the subscriber and Service we use it check that the types match, in this way we can better catch the issue.

### CODE REVIEW

**Review by floriantschopp — APPROVED** [review]

File: modules/random/include/hephaestus/random/random_object_creator.h [hunk 1]
```diff
@@ -153,12 +153,19 @@ namespace internal {
 /// Generate a random string of characters, including special case characters and numbers.
 template <StringType T>
 [[nodiscard]] auto random(std::mt19937_64& mt, std::optional<size_t> fixed_size = std::nullopt,
-                          bool allow_empty = true) -> T {
+                          bool allow_empty = true, bool lower_characters_only = false) -> T {
```
Comment (lhruby) [inline modules/random/include/hephaestus/random/random_object_creator.h:156]: Options for string creation start to sum up. What about adding a `RandomStringConfig` to bundle them?

**Review by lhruby — APPROVED** [review]
lgtm

File: modules/random/include/hephaestus/random/random_object_creator.h (hunk 1 above)
Comment (filippobrizzi) [reply modules/random/include/hephaestus/random/random_object_creator.h:156]: in this way it is possible to have default values

Comment (lhruby) [reply modules/random/include/hephaestus/random/random_object_creator.h]:
```
struct RandomStringConfig {
  std::optional<size_t> fixed_size = std::nullopt,
  bool allow_empty = true;
  bool lower_characters_only = false;
} RANDOM_STRING_CONFIG_DEFAULT; 

[[nodiscard]] auto random(std::mt19937_64& mt, RandomStringConfig config = RANDOM_STRING_CONFIG_DEFAULT) -> T {
```
The issue with default values is that you can only omit arguments in order, so you can e.g. not say `random(mt, use_only_lower_case_characters)`, so default values are limited. A config does not make things worse.

Comment (lhruby) [reply modules/random/include/hephaestus/random/random_object_creator.h]: By the way, this should be `lower_case_characters_only` or `force_lower_case_characters`.

Comment (filippobrizzi) [reply modules/random/include/hephaestus/random/random_object_creator.h]: I would resolve this, and address this in a follow up when we may want to add more options

**Pull request merged**